deltachat-ios/AppDelegate.swift

```diff
@@ -382,7 +388,16 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
             if !self.appIsInForeground() {
                 self.dcContext.stopIo()
             }
-            completionHandler(.newData)
+
+            // to avoid 0xdead10cc exceptions, scheduled jobs need to be done before we get suspended;
+            // we increase the probabilty that this happens by adding completionHandler()
+            // to the end of both dispatch queues.  
+            DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
```

Comment:
What is the purpose of the +1 seconds here and  [here](https://github.com/deltachat/deltachat-ios/pull/1210/files#diff-fc4377fbb4c60bfc9a11f53fd448de91bbf34787141509ddb85242515fea9e23R209?)

Reply:
just to give things a bit more time, to mitigate race conditions. might be that sth. gets dispatched from global to main queue, we would not catch that. 

there was a 3 second timeout before, but that appears too long.

Reply:
another thing: according to [developer.apple.com](https://developer.apple.com/documentation/dispatch/dispatchqueue/2300077-global), `DispatchQueue.global(qos:)` may return different queues, so, the additional second may catch some cases where other queues are used we are not checking. 

but it might be advantageous to first do swap the two schedules: background may easily schedule to main, the other way round, probably not so often. also, the 1 second is always added _after_ background finishes and does not run concurrently.